src/app/routines/publicRoutines/routineListItem/RoutineListItem.tsx

```diff
@@ -3,34 +3,98 @@ import { ListItem, ListItemText, Typography } from '@mui/material';
+  const {
+    id = '',
+    name = 'No name',
+    description = 'No description',
+    likesCount = 0,
+    isLikedByCurrentUser = false,
+  } = routine;
   const navigate = useNavigate();
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+
+  const { mutate: likeRoutine } = useMutation({
+    mutationFn: (id: string) => like(id, 'POST'),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['public-routines'] });
+    },
+    onError: () => {
+      setSnackbarOpen(true);
+    },
+  });
+
+  const { mutate: deleteRoutine } = useMutation({
+    mutationFn: (id: string) => like(id, 'DELETE'),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['public-routines'] });
+    },
+    onError: () => {
+      setSnackbarOpen(true);
+    },
+  });
+
+  const handleLikeClick = () => {
+    if (isLikedByCurrentUser) {
+      deleteRoutine(id);
+    } else {
+      likeRoutine(id);
+    }
+  };
+
+  const handleSnackbarClose = (_?: React.SyntheticEvent | Event, reason?: string) => {
+    if (reason && reason !== 'timeout') {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
 
   const openRoutineDetails = (routineId: string) => {
     const url = AppRoutes.ROUTINE_DETAILS.replace(':routineId', routineId);
     navigate(url);
   };
 
   return (
-    <div className='public-routine-list-item'>
-      <ListItem alignItems='flex-start' key={id}>
-        <ListItemText
-          onClick={() => openRoutineDetails(id)}
-          primary={name}
-          primaryTypographyProps={{
-            fontWeight: 'bold',
-          }}
-          className='public-routine-list-item__name'
-          secondary={
-            <Typography component='span' variant='body2' className='public-routine-list-item__description'>
-              {description}
-            </Typography>
-          }
-        />
-      </ListItem>
-    </div>
+    <>
+      <div>
+        <ListItem alignItems='flex-start' key={id} className='public-routine-list-item'>
+          <ListItemText
+            onClick={() => openRoutineDetails(id)}
+            primary={name}
+            primaryTypographyProps={{
+              fontWeight: 'bold',
+            }}
+            className='public-routine-list-item__name'
+            secondary={
+              <Typography component='span' variant='body2' className='public-routine-list-item__description'>
+                {description}
+              </Typography>
+            }
+          />
+          <div className='public-routine-list-item__likes' onClick={handleLikeClick}>
+            <div className='public-routine-list-item__likes-number'>
+              <p>{likesCount}</p>
+            </div>
+            <div className={`likes-icon ${isLikedByCurrentUser ? 'active' : ''}`}>
+              <ThumbUpOffAltIcon fontSize='small' />
+            </div>
+          </div>
+        </ListItem>
+      </div>
+      <AppAlert
```

Comment:
It would be better to add text message and severity as variable, if we add anything in the future its not reusable

Reply:
you want me to change the AppAlert component?